### PR TITLE
[opentitantool] HyperDebug refactoring, and control SPI speed

### PR DESCRIPTION
**Pull request by jesultra**
Introduce convenience functions for doing textual request/response with HyperDebug hardware.  Then use these methods to get and set SPI speed (supported on recent HyperDebug firmware).

### CODE REVIEW

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs [hunk 1]
```diff
@@ -268,8 +268,77 @@ pub struct Inner {
...
+
+    /// Send a command to HyperDebug firmware, expecting to receive a single line of output.  Any
+    /// more or less output will be reported through an `Err()` return.
+    pub fn cmd_one_line_output(&self, cmd: &str) -> Result<String> {
+        let mut result: Option<String> = None;
+        let mut unexpected_output: bool = false;
+        self.execute_command(cmd, |line| {
+            if unexpected_output {
+                // Third or subsequent line, report it.
+                log::warn!("Unexpected HyperDebug output: {}\n", line);
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/transport/hyperdebug/mod.rs:295]: Shouldn't this (and the `warn`s below) be `log::error!(...)`?

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs [hunk 2]
```diff
@@ -268,8 +268,77 @@ pub struct Inner {
 }
 
 impl Inner {
+    /// Send a command to HyperDebug firmware, expecting to receive no output.  Any output will be
+    /// reported through an `Err()` return.
+    pub fn cmd_no_output(&self, cmd: &str) -> Result<()> {
+        let mut unexpected_output: bool = false;
+        self.execute_command(cmd, |line| {
+            log::warn!("Unexpected HyperDebug output: {}\n", line);
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/transport/hyperdebug/mod.rs:276]: Should this be `log::error!(...)`?

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs [hunk 3]
```diff
@@ -268,8 +268,77 @@ pub struct Inner {
...
+    /// more or less output will be reported through an `Err()` return.
+    pub fn cmd_one_line_output_match<'a>(
+        &self,
+        cmd: &str,
+        regex: &Regex,
+        buf: &'a mut String,
+    ) -> Result<regex::Captures<'a>> {
+        *buf = self.cmd_one_line_output(cmd)?;
+        let Some(captures) = regex.captures(buf) else {
+            log::warn!("Unexpected HyperDebug output: {}\n", buf);
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/transport/hyperdebug/mod.rs:332]: `log::error!` here as well?

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs (hunk 1 above)
Comment (jesultra) [reply sw/host/opentitanlib/src/transport/hyperdebug/mod.rs]: Done

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs (hunk 2 above)
Comment (jesultra) [reply sw/host/opentitanlib/src/transport/hyperdebug/mod.rs]: Done

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs (hunk 3 above)
Comment (jesultra) [reply sw/host/opentitanlib/src/transport/hyperdebug/mod.rs]: Done

**Review by cfrantz — APPROVED** [review]

**Pull request merged**